src/data/TrackedEntityD2Repository.ts

```diff
@@ -0,0 +1,119 @@
+import _ from "lodash";
+import { D2Api } from "types/d2-api";
+import { Async } from "domain/entities/Async";
+
+import { getInChunks } from "./dhis2-utils";
+import { Stats } from "domain/entities/Stats";
+import {
+    TrackedEntityFilterParams,
+    TrackedEntityRepository,
+} from "domain/repositories/TrackedEntityRepository";
+import { TrackedEntity } from "domain/entities/TrackedEntity";
+import { D2TrackedEntity, ProgramsD2Repository } from "./ProgramsD2Repository";
+
+const initialStats = {
+    recordsSkipped: [],
+    errorMessage: "",
+    created: 0,
+    ignored: 0,
+    updated: 0,
+};
```

Comment:
This object is repeated in 2 repos. Also, we have some logic, `stats.reduce`.  Conclusion: Let's create a class Stats, with all we need (empty value, as static method, the accumulation)